### DISCUSSION TITLE
Yazi displays images and PDFs in binary format only

### DISCUSSION THREAD

**Discussion by AsternA** [sxyazi/yazi#1491] [1. Q&A]
Hey Guys,
I recently installed Yazi on MacOS from Homebrew:
brew install yazi ffmpegthumbnailer sevenzip jq poppler fd ripgrep fzf zoxide imagemagick font-symbols-only-nerd-font
it was working fine until I went to my images folder.
On the first image a window popped up asking for permission, I don't remember permission for what, I clicked OK and the image showed.
on the second image the same pop up window popped up and I clicked "don't ask me again" but this time I clicked DENY. 
Ever since then all my images are displayed in binary format and I can't do anything about it.
I have tried uninstalling it and reinstalling it but that didn't work.
Any ideas how to get it to display images again?
thanks

**Comment by AnirudhG07** [comment]
Was there app name or some other name you remember, present in the popup? It must some security issues you would have to change. If you could tell some name, ill look around for it. I never got such popups.

**Reply by AsternA** [reply]
Unfortunately I do not remember anything. It was hasty.
Just to be clear, when in Yazi the image preview doesn't work, if I clicked on 'o' then it opens the image / pdf in a separate window (and app) and it works just fine

**Reply by AnirudhG07** [reply]
right. but that's not ideal. Have you looked into Photos related Privacy Security in Settings?

**Reply by AnirudhG07** [reply]
Go to Setting > Privacy and Security > Photos, check the permissions there, I have given my terminal permission to see the photos

**Reply by AsternA** [reply]
![image](https://github.com/user-attachments/assets/0fb9473e-bceb-4db2-b464-9d7d51509334)

**Reply by AsternA** [reply]
I think this talks about the Photos 'app' not photos and pictures in general

**Reply by AnirudhG07** [reply]
yeah true. Ill look around for some permission issues. Worst case way to reset all app permissions, which is a bad idea imo.

**Comment by lpnh** [comment]
I'm not a macOS user, but what if you try giving Yazi 'Full Disk Access'? I'm suggesting this because it sounds like it might be related to a directory permission issue.

**Reply by AnirudhG07** [reply]
do you think running yazi with sudo would help?

**Reply by AsternA** [reply]
running sudo yazi reproduces the same results - images are binary

**Reply by AsternA** [reply]
@lpnh Yazi is running inside the terminal emulator iTerm2.
I gave iTerm2 full disk access but that didn't solve the issue :(

**Reply by AnirudhG07** [reply]
is there any other image viewer? maybe if its the preview we have to change, that could work

**Reply by AsternA** [reply]
Running sudo Yazi inside Tmux makes the image preview work!
I can see images and pdf previews

**Reply by AnirudhG07** [reply]
interesting

**Reply by AsternA** [reply]
verified again:
In TMUX session -> sudo yazi works, yazi does not
outside TMUX -> sudo yazi & yazi do not work

**Reply by AnirudhG07** [reply]
does imgcat work for you, since u are using iterm, its part of shell integrations

**Reply by AsternA** [reply]
It does not.
I went here: 
https://iterm2.com/documentation-images.html
copied the bash file, made it executable on my machine:
and this is the output:

![Screenshot 2024-08-15 at 14 49 39](https://github.com/user-attachments/assets/85eae2b4-596a-4e36-8c09-408f95b31eb1)

**Reply by AsternA** [reply]
Update: If outside of TMUX - imgcat displays the image in binary format (just like yazi)

**Reply by lpnh** [reply]
It really seems like a permission issue. I suspect it's related to one of the adapters. To check the environment, you can run `yazi --debug` in both cases and share only the `Emulator` and `Adapter` outputs here, like this:

```
yazi --debug

Emulator
    Emulator.via_env: ("foot", "")
    Emulator.via_csi: Ok(Foot)
    Emulator.detect : Foot

Adapter
    Adapter.matches: Sixel
```

**Reply by AsternA** [reply]
`Yazi
    Version: 0.3.0 (Homebrew 2024-08-01)
    Debug  : false
    OS     : macos-aarch64 (unix)

Ya
    Version: 0.3.0

Emulator
    Emulator.via_env: ("xterm-256color", "iTerm.app")
    Emulator.via_csi: Ok(Iterm2)
    Emulator.detect : Iterm2

Adapter
    Adapter.matches: Iterm2

Desktop
    XDG_SESSION_TYPE: None
    WAYLAND_DISPLAY : None
    DISPLAY         : None

SSH
    shared.in_ssh_connection: false

WSL
    /proc/sys/fs/binfmt_misc/WSLInterop: false

Variables
    SHELL              : Some("/bin/zsh")
    EDITOR             : None
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None
    ZELLIJ_SESSION_NAME: None

Text Opener
    default: Some(Opener { run: "${EDITOR:=vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:=vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

tmux
    TMUX   : false
    Version: 3.4

Dependencies
    file             : 5.41
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.2
    magick           : 7.1.1-36
    fzf              : 0.54.3
    fd               : 10.1.0
    rg               : 14.1.0
    chafa            : No such file or directory (os error 2)
    zoxide           : 0.9.4
    7z               : No such file or directory (os error 2)
    7zz              : 24.08
    jq               : 1.7.1`

**Reply by AsternA** [reply]
Here is the diff when running yazi through tmux.
on the left yazi --debug without TMUX
on the right yazi --debug with TMUX
![image](https://github.com/user-attachments/assets/c89beda8-8939-4cd4-a330-d59ffb27a249)

**Reply by AsternA** [reply]
When inside tmux running sudo yazi --debug:

```
Yazi
    Version: 0.3.0 (Homebrew 2024-08-01)
    Debug  : false
    OS     : macos-aarch64 (unix)

Ya
    Version: 0.3.0

Emulator
    Emulator.via_env: ("screen-256color", "")
    Emulator.via_csi: Ok(Unknown([Sixel]))
    Emulator.detect : Unknown([Sixel])

Adapter
    Adapter.matches: Sixel

Desktop
    XDG_SESSION_TYPE: None
    WAYLAND_DISPLAY : None
    DISPLAY         : None

SSH
    shared.in_ssh_connection: false

WSL
    /proc/sys/fs/binfmt_misc/WSLInterop: false

Variables
    SHELL              : Some("/bin/sh")
    EDITOR             : None
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None
    ZELLIJ_SESSION_NAME: None

Text Opener
    default: Some(Opener { run: "${EDITOR:=vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:=vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

tmux
    TMUX   : false
    Version: 3.4

Dependencies
    file             : 5.41
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.2
    magick           : 7.1.1-36
    fzf              : 0.54.3
    fd               : 10.1.0
    rg               : 14.1.0
    chafa            : No such file or directory (os error 2)
    zoxide           : 0.9.4
    7z               : No such file or directory (os error 2)
    7zz              : 24.08
    jq               : 1.7.1
```

**Reply by sxyazi** [reply]
What's your iTerm2 version? Is it the latest?

**Reply by AnirudhG07** [reply]
> Update: If outside of TMUX - imgcat displays the image in binary format (just like yazi)

Try installing shell integrations properly. You should be able to use imgcat without doing `./imgcat image.png`. See on their website on how to do it.

**Reply by AsternA** [reply]
TMUX is latest version.

**Reply by AsternA** [reply]
Installing Shell integrations went well:
Iterm2 -> Install Shell Integrations 
Still imgcat does not work :(

**Reply by AsternA** [reply]
activating debug log on iTerm2 and trying imgcat I find the following line:

`|   <iTermImageView: 0x126e575f0 image=(null) hidden=1 alpha=1> frame=NSRect: {{0, 0}, {1460, 765}} hidden=YES alphaValue=1.00 autoresizing=h,w,subviews autolayout=No tracking_areas=none
`
For some reason it says image is null

**Reply by AsternA** [reply]
One last update.
I installed WezTerm. Imgcat and yazi image previewer work out of the box.
So this is probably a terminal permission issue

**Reply by AnirudhG07** [reply]
So things are fixed now? Since you are using westerm and not iterm.

**Reply by AsternA** [reply]
I guess so.
I'll do one more test over the weekend. Uninstall iTerm and reinstall maybe that will reset its permissions.

**Reply by AsternA** [reply]
Deleted and reinstalled iTerm2. The issue remained the same.
Moved to wezterm. 
Thanks everyone for your help
